packages/react/src/hooks/useBlockNote.ts

```diff
@@ -25,15 +25,16 @@ export const useBlockNote = <BSchema extends BlockSchema = DefaultBlockSchema>(
   options: Partial<BlockNoteEditorOptions<BSchema>> = {},
   deps: DependencyList = []
 ): BlockNoteEditor<BSchema> => {
-  const editorRef = useRef<BlockNoteEditor<BSchema>>(initEditor(options));
+  const editorRef = useRef<BlockNoteEditor<BSchema>>();
 
-  useMemo(() => {
+  const ret = useMemo(() => {
     if (editorRef.current) {
-      editorRef.current && editorRef.current._tiptapEditor.destroy();
+      editorRef.current._tiptapEditor.destroy();
     }
 
     editorRef.current = initEditor(options);
-  }, [options, ...deps]);
+    return editorRef.current;
+  }, [deps, options]); //eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
this options dependency is still wrong

Reply:
I'm confused on how to do this - I tried another approach:
```ts
export const useBlockNote = <BSchema extends BlockSchema = DefaultBlockSchema>(
  options: Partial<BlockNoteEditorOptions<BSchema>> = {},
  deps: DependencyList = []
): BlockNoteEditor<BSchema> => {
  const editorRef = useRef<BlockNoteEditor<BSchema>>();
  const newDeps = [...deps, options];

  const ret = useMemo(() => {
    ...
  }, newDeps);

  return ret;
};
```
But this also gives me a linting error that options is not in deps.

Reply:
there's //eslint-disable-line react-hooks/exhaustive-deps right?

Reply:
Yeah, but I'm still confused. I get that now we just pass an already created array to deps so React doesn't have to keep recreating it, but how come it's not an issue anywhere else we use hooks? Since pretty much everywhere else, we do declare the array in deps.